pkg/registry/storage/volumeattachment/strategy_test.go

```diff
@@ -54,24 +63,135 @@ func TestVolumeAttachmentStrategy(t *testing.T) {
 		t.Errorf("unexpected error validating %v", errs)
 	}
 
-	newVolumeAttachment := &storage.VolumeAttachment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: "valid-attachment-2",
-		},
-		Spec: storage.VolumeAttachmentSpec{
-			Attacher: "valid-attacher-2",
-			Source: storage.VolumeAttachmentSource{
-				PersistentVolumeName: &pvName,
-			},
-			NodeName: "valid-node-2",
-		},
+	// Create with status should drop status
+	statusVolumeAttachment := volumeAttachment.DeepCopy()
+	statusVolumeAttachment.Status = storage.VolumeAttachmentStatus{Attached: true}
+	Strategy.PrepareForCreate(ctx, statusVolumeAttachment)
+	if !apiequality.Semantic.DeepEqual(statusVolumeAttachment, volumeAttachment) {
+		t.Errorf("unexpected objects difference after creating with status: %v", diff.ObjectDiff(statusVolumeAttachment, volumeAttachment))
 	}
 
+	// Update of spec is disallowed
+	newVolumeAttachment := volumeAttachment.DeepCopy()
+	newVolumeAttachment.Spec.NodeName = "valid-node-2"
+
 	Strategy.PrepareForUpdate(ctx, newVolumeAttachment, volumeAttachment)
 
 	errs = Strategy.ValidateUpdate(ctx, newVolumeAttachment, volumeAttachment)
 	if len(errs) == 0 {
 		t.Errorf("Expected a validation error")
 	}
 
+	// modifying status should be dropped
+	statusVolumeAttachment = volumeAttachment.DeepCopy()
+	statusVolumeAttachment.Status = storage.VolumeAttachmentStatus{Attached: true}
+
+	Strategy.PrepareForUpdate(ctx, statusVolumeAttachment, volumeAttachment)
+
+	if !apiequality.Semantic.DeepEqual(statusVolumeAttachment, volumeAttachment) {
+		t.Errorf("unexpected objects difference after modfying status: %v", diff.ObjectDiff(statusVolumeAttachment, volumeAttachment))
+	}
+}
+
+func TestVolumeAttachmentStatusStrategy(t *testing.T) {
+	ctx := genericapirequest.WithRequestInfo(genericapirequest.NewContext(), &genericapirequest.RequestInfo{
+		APIGroup:   "storage.k8s.io",
+		APIVersion: "v1",
+		Resource:   "volumeattachments",
+	})
+
+	volumeAttachment := getValidVolumeAttachment("valid-attachment")
+
+	// modifying status should be allowed
+	statusVolumeAttachment := volumeAttachment.DeepCopy()
+	statusVolumeAttachment.Status = storage.VolumeAttachmentStatus{Attached: true}
+
+	expectedVolumeAttachment := statusVolumeAttachment.DeepCopy()
+	StatusStrategy.PrepareForUpdate(ctx, statusVolumeAttachment, volumeAttachment)
+	if !apiequality.Semantic.DeepEqual(statusVolumeAttachment, expectedVolumeAttachment) {
+		t.Errorf("unexpected objects differerence after modifying status: %v", diff.ObjectDiff(statusVolumeAttachment, expectedVolumeAttachment))
+	}
+
+	// modifying spec should be dropped
+	newVolumeAttachment := volumeAttachment.DeepCopy()
+	newVolumeAttachment.Spec.NodeName = "valid-node-2"
```

Comment:
add a label, annotation, finalizer, and ownerRef as well and make sure they are ignored